### DISCUSSION TITLE
ESP32 Sandeep Lora library (callback mode) and Bodmer TFT-eSPI library crash?

### DISCUSSION THREAD

**Discussion by peringek** [Bodmer/TFT_eSPI#1185] [Q&A - General]
I use ESP32/vscode/platformio/arduino with my own hw. I have a RFM95 lora radio who shares SPI lines with my ILI9341 240X320 LCD. MISO is not connected on the display and the radio and LCD has different CS lines to the ESP. This display also has touch screen witch I have not connected. 
When polling the radio for received packages I found that the radio would some times stop receiving. This happened at random times and my system could work flawlessly for hours - but also stop after minutes. The display also worked flawlessly and my updateLCD routine took around 60 mSecs to complete. This was done every second.
Unable to make the connection stable I decided to use callbacks for LORA messages. My IRQ routine would set a flag and the reading of Lora data be done later. 
Then I started to get lots of resets due to WDT timeout. 
![image](https://user-images.githubusercontent.com/64654302/117942235-2283b080-b30b-11eb-9140-1322698cb67f.png)
During boot I notice that when the Lora radio or the LCD is initialized an error message is produced:
**[E][esp32-hal-cpu.c:93] addApbChangeCallback(): duplicate func=401049E0 arg=3FFBF140**
In this example LCD is initialized after the radio so the error occurs on tft.init();
If I would initialize Lora after LCD the error would occur during Lora initialization. 
This is what my board prints out during boot:
...
_Motorstatus at start=0
flagOPEN at start=0
flagCLOSE at start=0
Operation Mode = REMOTE
LoRa Receiver Callback
Starting LoRa finished
LCD Setup Start...
[E][esp32-hal-cpu.c:93] addApbChangeCallback(): duplicate func=401049E0 arg=3FFBF140
LCD Setup Stopp...
- Clear LCD - LCD Updated!
 TX Counts: 1 from[210]  to[213]  id[1]  cmd[0]  reg[3]  0[100]  1[22]  2[33]  3[44]    Tx/rX Diff:1
 RX Counts: 1 from[213]  to[210]  id[1]  cmd[10]  reg[3]  0[55]  1[0]  2[0]  3[60]    Missing: 0_
...

So now the program is running sending and receiving Lora. If I now try to go through my menues on the LCD the program will throw a "Guru Meditation Error: Core  1 panic'ed (Interrupt wdt timeout on CPU1)". 
If I leave it alone it will work fine for a while before it crashes.

Is this problem one that it is possible to solve ?

**Comment by Bodmer** [comment]
See [this issue](https://github.com/Bodmer/TFT_eSPI/issues/429), it may be relevant.

**Reply by Bodmer** [reply]
See also [this one](https://github.com/Bodmer/TFT_eSPI/issues/987).

**Reply by peringek** [reply]
Thank you ! I will try :)
